pkg/apis/core/v1/conversion.go

```diff
@@ -539,3 +546,146 @@ func Convert_core_PersistentVolumeSpec_To_v1_PersistentVolumeSpec(in *core.Persi
+		if ctr.SecurityContext != nil {
+			ctrInField = ctr.SecurityContext.SeccompProfile
+		}
+
+		// Find the annotation value
+		annotationKey := v1.SeccompContainerAnnotationKeyPrefix + ctr.Name
+		var annotationValue string
+		for key, value := range in.Annotations {
+			if key == annotationKey {
+				annotationValue = value
+				break
+			}
+		}
+
+		profile, err := syncSeccompAnnotationAndFieldForPod(
+			annotationKey, annotationValue, ctrInField, out,
+		)
+		if err != nil {
+			return errors.Wrapf(err, "synchronizing annotations and security context field for container: %+v", ctr)
+		}
+
+		if profile != nil {
+			// Ensure that the output security context is not nil and set the new
+			// profile
+			if out.Spec.Containers[i].SecurityContext == nil {
+				out.Spec.Containers[i].SecurityContext = &v1.SecurityContext{}
+			}
+			out.Spec.Containers[i].SecurityContext.SeccompProfile = profile
+		}
+	}
+
+	return nil
+}
+
+func syncSeccompAnnotationAndFieldForPod(annotationKey, annotationValue string, field *core.SeccompProfile, pod *v1.Pod) (*v1.SeccompProfile, error) {
+	// If no seccomp annotations or fields are specified, then no further
+	// action is necessary.
+	if field == nil && annotationValue == "" {
+		return nil, nil
+	}
+
+	// If only seccomp fields are specified, add the corresponding annotations.
+	// This ensures that the fields are enforced even if the node version
+	// trails the API version
+	if field != nil && annotationValue == "" {
+		switch field.Type {
+		case core.SeccompProfileUnconfined:
+			pod.Annotations[annotationKey] = v1.SeccompProfileNameUnconfined
+
+		case core.SeccompProfileTypeDefault:
+			pod.Annotations[annotationKey] = v1.SeccompProfileRuntimeDefault
+
+		case core.SeccompProfileLocalhost:
+			if field.LocalhostProfile == nil {
+				return nil, errors.Errorf("localhost profile is nil")
+			}
+			pod.Annotations[annotationKey] = v1.SeccompLocalhostProfileNamePrefix + *field.LocalhostProfile
+		}
+		return nil, nil
+	}
+
+	// If only seccomp annotations are specified, copy the values into the
+	// corresponding fields. This ensures that existing applications continue
+	// to enforce seccomp, and prevents the kubelet from needing to resolve
+	// annotations & fields.
+	if field == nil && annotationValue != "" {
+		profile := &v1.SeccompProfile{}
+		if annotationValue == v1.SeccompProfileNameUnconfined {
+			profile.Type = v1.SeccompProfileUnconfined
+
+		} else if annotationValue == v1.SeccompProfileRuntimeDefault {
+			profile.Type = v1.SeccompProfileTypeDefault
+
+		} else if strings.HasPrefix(annotationValue, v1.SeccompLocalhostProfileNamePrefix) {
+			localhostProfile := strings.TrimPrefix(annotationValue, v1.SeccompLocalhostProfileNamePrefix)
+			profile.Type = v1.SeccompProfileLocalhost
+			profile.LocalhostProfile = &localhostProfile
+		}
+		return profile, nil
+	}
+
+	// If both seccomp annotations and fields are specified, the values must
+	// match.
+	if field != nil && annotationValue != "" {
```

Comment:
this is enforced in validation, don't need to enforce this here

Reply:
Okay, removed the whole block.